File: new-resource-detector/src/test/java/nl/haploid/octowight/service/DirtyResourceProducerServiceIT.java
package nl.haploid.octowight.service;

import nl.haploid.octowight.AbstractIT;
import nl.haploid.octowight.TestData;
import nl.haploid.octowight.registry.data.ResourceRoot;
import org.apache.kafka.clients.producer.RecordMetadata;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.concurrent.Future;

import static org.junit.Assert.assertNotNull;

public class DirtyResourceProducerServiceIT extends AbstractIT {

	@Autowired
	private DirtyResourceProducerService service;

	@Test
	public void testSendDirtyResource() throws Exception {
		final ResourceRoot resourceRoot = TestData.resourceRoot(555l);
		final Future<RecordMetadata> future = service.sendDirtyResource(resourceRoot);
		assertNotNull(future.get());
	}
}

File: event-consumer/src/test/scala/nl/haploid/octowight/service/DirtyResourceProducerServiceIT.scala
package nl.haploid.octowight.service

import java.util.concurrent.TimeUnit

import nl.haploid.octowight.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class DirtyResourceProducerServiceIT extends AbstractIT {
  @Autowired private[this] val dirtyResourceProducerService: DirtyResourceProducerService = null

  "Dirty resource producer" should "send dirty resource" in {
    val resourceRoot = TestData.resourceRoot(555l)
    val future = dirtyResourceProducerService.sendDirtyResource(resourceRoot)
    Option(future.get(5, TimeUnit.SECONDS)) should not be None
  }
}
